components/core/src/clp/Grep.hpp

```diff
@@ -12,6 +13,122 @@
 #include "streaming_archive/reader/File.hpp"
 
 namespace clp {
+
+/**
+ * Represents a logtype that would match the given search query. The logtype is a sequence
+ * containing values, where each value is either a static character or an integers representing
+ * a variable type id. Also indicates if an integer/float variable is potentially in the dictionary
+ * to handle cases containing wildcards. Note: long float and integers that cannot be encoded do not
+ * fall under this case, as they are not potentially, but definitely in the dictionary, so will be
+ * searched for in the dictionary regardless.
+ */
+class QueryLogtype {
+public:
+    std::vector<std::variant<char, int>> m_logtype;
+    std::vector<std::string> m_search_query;
+    std::vector<bool> m_is_potentially_in_dict;
+    std::vector<bool> m_var_has_wildcard;
+
+    /**
+     * Append a logtype to the current logtype.
+     * @param suffix
+     */
+    auto append_logtype(QueryLogtype& suffix) -> void {
```

Comment:
The symbols in this class aren't in the required [declaration order](https://google.github.io/styleguide/cppguide.html#Declaration_Order).

Reply:
Should be in the right order now. Also moved longer function definitions into cpp.